src/DynamoCore/Graph/Nodes/ZeroTouch/DSFunctionBase.cs

```diff
@@ -40,11 +40,15 @@ public override bool IsConvertible
 
         public override IdentifierNode GetAstIdentifierForOutputIndex(int outputIndex)
         {
-            return Controller.ReturnKeys != null && Controller.ReturnKeys.Any()
-                ? base.GetAstIdentifierForOutputIndex(outputIndex)
-                : (OutPortData.Count == 1
-                    ? AstIdentifierForPreview
-                    : base.GetAstIdentifierForOutputIndex(outputIndex));
+            if (Controller.ReturnKeys != null && Controller.ReturnKeys.Any())
+            {
+                string id = AstIdentifierBase + "_out" + outputIndex;
+                return AstFactory.BuildIdentifier(id);
+            }
+            else
+            {
+                return base.GetAstIdentifierForOutputIndex(outputIndex); 
+            }
```

Comment:
The original logic is, if there are two or more outport, then call `base.GetAstIdentifierForOutputIndex()` otherwise just use `AstIdentifierForPreview`. The change is, if there are some return keys (which means the return value is from dictionary), we always add index to the identifier. So yes, it is intended change.